src/helpers/sso-helper.ts

```diff
@@ -3,34 +3,40 @@
  * See LICENSE in the project root for license information.
  */
 
-import { dialogFallback } from "./fallbackauthhelper";
-import { handleClientSideErrors, makeGraphApiCall, showMessage } from "office-addin-sso";
+import { dialogFallback } from "./fallbackauthdialog";
+import { callGetUserData } from "./middle-tier-calls";
+import { showMessage } from "./message-helper";
+import { handleClientSideErrors } from "./error-handler";
 
 /* global OfficeRuntime */
 
-let retryGetAccessToken = 0;
+let retryGetMiddletierToken = 0;
 
-export async function getGraphData(callback): Promise<void> {
+export async function getUserData(callback): Promise<void> {
   try {
-    let bootstrapToken: string = await OfficeRuntime.auth.getAccessToken({ allowSignInPrompt: true });
-    let response: any = await makeGraphApiCall(bootstrapToken);
-    if (response.claims) {
+    let middletierToken: string = await OfficeRuntime.auth.getAccessToken({
+      allowSignInPrompt: true,
+      allowConsentPrompt: true,
+      forMSGraphAccess: true,
+    });
+    let response: any = await callGetUserData(middletierToken);
+    if (!response) {
+      return Promise.reject();
+    } else if (response.claims) {
       // Microsoft Graph requires an additional form of authentication. Have the Office host
       // get a new token using the Claims string, which tells AAD to prompt the user for all
       // required forms of authentication.
-      let mfaBootstrapToken: string = await OfficeRuntime.auth.getAccessToken({
+      let mfaMiddletierToken: string = await OfficeRuntime.auth.getAccessToken({
         authChallenge: response.claims,
       });
-      response = makeGraphApiCall(mfaBootstrapToken);
+      response = callGetUserData(mfaMiddletierToken);
     }
 
     // AAD errors are returned to the client with HTTP code 200, so they do not trigger
     // the catch block below.
     if (response.error) {
       handleAADErrors(response, callback);
     } else {
-      // makeGraphApiCall makes an AJAX call to the MS Graph endpoint. Errors are caught
-      // in the .fail callback of that call
       callback(response);
       Promise.resolve();
```

Comment:
Again, async function, so why not just return the result rather than use the Promise pattern?